src/org/openstreetmap/josm/gui/NavigatableComponent.java

```diff
@@ -149,6 +154,51 @@ protected static void fireZoomChanged() {
         });
     }
 
+    private final CopyOnWriteArrayList<PrimitiveHoverListener> primitiveHoverListeners = new CopyOnWriteArrayList<>();
+    // Allow primitive to be garbage collected by using a weak reference
+    private WeakReference<OsmPrimitive> previousHoveredPrimitive = new WeakReference<>(null);
```

Comment:
Wasn't entirely sure if it would be worth it to do it this way but I added it to be on the safe side. I don't know if there are some edge cases where primitives can get really large, but I guess those really are edge cases and it shouldn't stick around for long in any case. Will change it back to a normal variable.

> If code users want to keep track of the previous object, that should probably be done in their code.

We need to keep track of the previous object anyway so that we can fire the hover events only when something changed.